### PR TITLE
Add prismatic reading support

### PR DESCRIPTION
**Pull request by ericpre**
This PR add new class in the emd reader to support prismatic files. It should also be more up to date with the recent version of the emd ncem format (https://emdatasets.com) and I have left the old one not to break the API. The reader is using the new class for reading and writing emd NCEM files.

### Description of the change
- add support for reading prismatic emd fild
- rename `dataset_name` to `dataset_path` in emd (NCEM) reader
- rename `dset_name` to `dataset_path` in usid reader for consistency
- fix transpose data of EMD/NCEM file.

### Progress of the PR
- [x] Change implemented,
- [x] update docstring,
- [x] update user guide,
- [x] add tests,
- [x] ready for review.

### CODE REVIEW

File: doc/user_guide/io.rst [hunk 1]
```diff
@@ -385,13 +385,13 @@ Extra saving arguments
...
+- ``compressor``: The compression codec, one of [`None`,`'zlib`',`'zstd'`, `'lz4'`].
   Defaults to `None`.
-- `clevel`: The compression level, an `int` from 1 to 9. Defaults to 1.
-- `n_threads`: The number of threads to use for `blosc` compression. Defaults to
+- ``clevel``: The compression level, an `int` from 1 to 9. Defaults to 1.
+- ``n_threads``: The number of threads to use for 'blosc' compression. Defaults to
   the maximum number of virtual cores (including Intel Hyperthreading)
   on your system, which is recommended for best performance. If \
-  `do_asyc = True` you may wish to leave one thread free for the
+  ``do_asyc = True`` you may wish to leave one thread free for the
```
Comment (tjof2) [inline doc/user_guide/io.rst]: Typo - surely do_async?

**Review by tjof2 — APPROVED** [review]

File: hyperspy/io_plugins/emd.py [hunk 2]
```diff
@@ -128,6 +130,9 @@ class EMD(object):
 
     def __init__(self, signals=None, user=None,
                  microscope=None, sample=None, comments=None):
+        warnings.warns("Direct instantiation of the EMD class is deprecated "
+                       "and will be removed in HyperSpy v2.0. Please use the "
+                       "`hs.load` function instead.")
```
Comment (tjof2) [inline hyperspy/io_plugins/emd.py]: Should this raise a `VisibleDeprecationWarning` like usual?

**Review by tjof2 — COMMENTED** [review]
One v.minor change then LGTM

File: hyperspy/io_plugins/emd.py (hunk 2 above)
Comment (ericpre) [reply hyperspy/io_plugins/emd.py]: Done!

**Comment by tjof2** [comment]
Hmm why the big change in codecov?

**Comment by ericpre** [comment]
This is because the reader is now using the new class `EMD_NCEM` instead of the deprecated `EMD`, which is not covered anymore by the tests.

**Comment by tjof2** [comment]
Makes sense.

**Pull request merged**